tests/refactor: make screen.ui use "linegrid" representation internally

bfredl opened the pull request:
#8221 took a short-cut when implementing the tests: screen.lua would translate the linegrid highlight ids back into the old per-cell attribute description.

Apart from cleaning up technical debt, this enables to check both rgb and cterm colors in the same expect(), which previously was needlessly restricted to ext_hlstate tests only. This will be needed for #10994 tests, but I though it would make sense to test this separately against master. It should cause no functional change, unless some tests does some thing weird, like terminal/helpers.lua which contained a duplicated highlight description.

bfredl committed:
tests/ui: make screen.lua use "linegrid" representation internally

PR #8221 took a short-cut when implementing the tests: screen.lua would
translate the linegrid highlight ids back into the old per-cell
attribute description.

Apart from cleaning up technical debt, this enables to check both rgb
and cterm colors in the same expect(), which previously was needlessly
restricted to ext_hlstate tests only.

bfredl commented:
Note to self: "UNEXPECTED" warnings could be improved.

bfredl committed:
tests/ui: cleanup illegitimate usages of "attr_ignore"

"attr_ignore" is an anti-pattern, with snapshot_util()
just include all the highlights already.

justinmk reviewed (approved):
`_rgb_cterm` seems like a contradictory name , but I assume I just need more time with the API here.

bfredl commented:
it means "rgb AND cterm" as opposed to the normal of only checking one of them. Happy to take naming suggestions. `multicolor`? `doublevision` ?

bfredl committed:
tests/ui: completely delete "attr_ignore" feature

All existing usages are ad-hoc/random/lazyness. Generating attribute
specifications is not hard since four years, just do it always.

bfredl committed:
tests/ui: remove unnecessary screen:detach()

It is perfectly fine and expected to detach from the screen just by
the UI disconnecting from nvim or exiting nvim. Just keep detach() in
screen_basic_spec, to get some coverage of the detach method itself.

This avoids hang on failure in many situations (though one could argue
that detach() should be "fast", or at least "as fast as resize",
which works in press-return already).

Never use detach() just to change the size of the screen, try_resize()
method exists for that specifically.

bfredl commented:
removed `after_each(screen:detach())` which is a weird cargo-culted anti-pattern. Nvim can detach an UI (and exit, in case of channel 1) perfectly fine by just disconnecting the channel. This probably saves a second or two in total, as well as avoids lots of potential hang-on-failure (when there was an unexpected error message).

bfredl commented:
Note: ended up deleting `attr_ignore` feature as there were no legitimate uses (i e no place where the ignore did anything for the meaning of the test). Generating highlight tests with `snapshot_util()` is not hard, just do it always. But the deletion is a separate commit, so we can restore it later if needed.
